sdk/core/azure-core/tests/async_tests/test_authentication_async.py

```diff
@@ -35,33 +44,45 @@ async def get_token(_):
 
     fake_credential = Mock(get_token=get_token)
     policies = [AsyncBearerTokenCredentialPolicy(fake_credential, "scope"), Mock(send=verify_authorization_header)]
-    pipeline = AsyncPipeline(transport=Mock(), policies=policies)
+    transport = Mock()
+    add_properties_to_transport(transport)
+    pipeline = AsyncPipeline(transport=transport, policies=policies)
 
-    await pipeline.run(HttpRequest("GET", "https://spam.eggs"), context=None)
+    await pipeline.run(request_type("GET", "https://spam.eggs"), context=None)
     assert get_token_calls == 1
 
-    await pipeline.run(HttpRequest("GET", "https://spam.eggs"), context=None)
+    await pipeline.run(request_type("GET", "https://spam.eggs"), context=None)
     # Didn't need a new token
     assert get_token_calls == 1
 
-
-async def test_bearer_policy_send():
+@pytest.mark.parametrize("request_type", [PipelineTransportHttpRequest, RestHttpRequest])
+async def test_bearer_policy_send(request_type, add_properties_to_transport):
     """The bearer token policy should invoke the next policy's send method and return the result"""
-    expected_request = HttpRequest("GET", "https://spam.eggs")
+    expected_request = request_type("GET", "https://spam.eggs")
     expected_response = Mock()
 
     async def verify_request(request):
-        assert request.http_request is expected_request
+        if hasattr(request_type, "content"):
```

Comment:
I use the attribute check of "content" to determine whether we're using pipline transport stuff, or rest stuff